apps/dgst.c

```diff
@@ -503,6 +503,8 @@ static void show_digests(const OBJ_NAME *name, void *arg)
     } else {
         BIO_printf(dec->bio, " ");
     }
+
+    EVP_MD_free(md);
```

Comment:
You need to be careful here.  md is allocated in two ways here:
1) EVP_md_fetch, which does need to be freed
2) EVP_get_digestbyname, which must _not_ be freed

You can't unilaterally free md, you have to record how it was allocated, and only free it if EVP_md_fetch returned the pointer

Reply:
Good to know :) I'll refactor ;)

Reply:
Addressed.